xcarchive/ios.go

```diff
@@ -210,17 +211,19 @@ func NewIosArchive(path string) (IosArchive, error) {
 
 	application := IosApplication{}
 	{
-		pattern := filepath.Join(path, "Products/Applications/*.app")
-		pths, err := filepath.Glob(pattern)
-		if err != nil {
-			return IosArchive{}, err
-		}
-
 		appPath := ""
-		if len(pths) > 0 {
-			appPath = pths[0]
+		if appRelativePathToProducts, found := applicationFromPlist(infoPlist); found {
+			appPath = filepath.Join(path, "Products", appRelativePathToProducts)
 		} else {
-			return IosArchive{}, fmt.Errorf("failed to find main app, using pattern: %s", pattern)
+			var err error
```

Comment:
this is only required because you are using scope  bumps, otherwise here you should have an existing instance of the err var. I think it is not a good idea to use scope bump just for readability. These blocks works like a miniature inline functionality, used in cases when you need to use same variable names in the same place.

https://medium.com/golangspec/scopes-in-go-a6042bb4298c